src/restic_fmt_plug.c

```diff
@@ -0,0 +1,340 @@
+/*
+ * This software is Copyright (c) 2020, Jürgen Hötzel <juergen at hoetzel.info>
+ * and it is hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_restic;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_restic);
+#endif
+
+#include <string.h>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#include "aes.h"
+#include "arch.h"
+#include "base64_convert.h"
+#include "common.h"
+#include "formats.h"
+#include "misc.h"
+#include "poly1305-donna/poly1305-donna.h"
+#include "yescrypt/sysendian.h"
+#include "yescrypt/yescrypt.h"
+
+#define FORMAT_NAME "Restic Repository"
+#define FORMAT_LABEL "restic"
+#define FORMAT_TAG "$restic$"
+#define TAG_LENGTH sizeof(FORMAT_TAG) - 1
```

Comment:
Not specific to this format, but ideally we'd have braces around the `sizeof(FORMAT_TAG) - 1`.